src/Components/Web.JS/src/Rendering/LogicalElements.ts

```diff
@@ -49,16 +53,21 @@ export function toLogicalRootCommentElement(start: Comment, end: Comment): Logic
   // |- *div
   // |- *component
   // |- *footer
-  if (!start.parentNode) {
+  const { start, end } = descriptor;
+  const parent = start.parentNode;
+
+  if (!parent) {
     throw new Error(`Comment not connected to the DOM ${start.textContent}`);
   }
 
-  const parent = start.parentNode;
   const parentLogicalElement = toLogicalElement(parent, /* allow existing contents */ true);
   const children = getLogicalChildrenArray(parentLogicalElement);
-  Array.from(parent.childNodes).forEach(n => children.push(n as unknown as LogicalElement));
```

Comment:
This was causing a bug where if two or more components get prerendered inside the same DOM element, the logical children array gets duplicate nodes. This results in weird behavior where components might swap places in the DOM after being activated.

I might be missing some context, but I'm not sure why it was done this way instead of only treating the 'start' and 'end' comments as logical elements. The content between them will get deleted soon after this anyway. This change didn't cause any problems in my testing, but maybe others have insight into the original thinking behind this.

Reply:
> This was causing a bug where if two or more components get prerendered inside the same DOM element, the logical children array gets duplicate nodes.

I totally see why that would have happened - the older logic simply failed to account for this situation.

However I would expect the solution here to be simply checking whether the existing parent is already configured as a logical element, and if so, not rebuilding its array of children. If it's already a LogicalElement, then we just have to update the `start` to have `start[logicalEndSiblingPropname] = end` so that when we're walking the siblings we automatically skip over the content nodes between start and end.

> I might be missing some context, but I'm not sure why it was done this way instead of only treating the 'start' and 'end' comments as logical elements

We can't only treat the start/end as logical elements because then the parent would not have an accurate list of its children. If someone called the LogicalElements APIs to mutate the parent's set of children in some way, it would get confused because the logical view wouldn't be in sync with the physical reality.

To keep things consistent, if the parent is being configured as a logical element, it needs an accurate list of children, which includes child nodes that are LogicalElements and ones that are not.